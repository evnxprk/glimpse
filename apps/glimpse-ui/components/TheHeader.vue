<template>
  <div>
    <!-- Mobile Menu -->
    <VNavigationDrawer v-model="sidebar" app temporary dark>
      <VList nav>
        <!-- Header -->
        <VListItem class="drawer-header">
          <img src="../assets/rpitv_logo_400.png" alt="Logo" class="tv-logo">
          <h3>RPI TV</h3>
        </VListItem>
        <VListItemSubtitle class="drawer-social">
          <a
            v-for="item in socials"
            v-if="!item.shouldDisplay || item.shouldDisplay()"
            :key="item.title"
            :href="item.path"
            :aria-label="item.title"
          >
            <font-awesome-icon :icon="['fab', item.icon]" size="lg" />
          </a>
        </VListItemSubtitle>

        <VDivider />

        <!-- Left menu items -->
        <VListItem
          v-for="item in leftMenuItems"
          v-if="!item.shouldDisplay || item.shouldDisplay()"
          :key="item.title"
          :to="item.path"
          class="drawer-item"
        >
          <VListItemAction>
            <font-awesome-icon :icon="['fad', item.icon]" class="drawer-icon" />
          </VListItemAction>

          <VListItemContent class="drawer-item-content">
            {{ item.title }}
          </VListItemContent>
        </VListItem>

        <VDivider />

        <!-- Right menu items -->
        <VListItem
          v-for="item in rightMenuItems"
          v-if="!item.shouldDisplay || item.shouldDisplay()"
          :key="item.title"
          :to="item.path"
          class="drawer-item"
        >
          <VListItemAction>
            <font-awesome-icon :icon="['fad', item.icon]" class="drawer-icon" />
          </VListItemAction>

          <VListItemContent class="drawer-item-content">
            {{ item.title }}
          </VListItemContent>
        </VListItem>
      </VList>
    </VNavigationDrawer>

    <!-- Desktop nav bar -->
    <VAppBar
      :color="scrollDistance > 0 || !$props.transparentAtTop ? this.$vuetify.theme.currentTheme.primary.darken1 : '#00000000'"
      app
      elevate-on-scroll
      dark
    >
      <span class="hidden-sm-and-up" aria-label="open navigation drawer">
        <font-awesome-icon :icon="['fal', 'bars']" @click="sidebar = !sidebar" class="drawer-icon" />
      </span>

      <VToolbarTitle class="app-nav-header">
        <img src="../assets/rpitv_logo_400.png" alt="Logo" class="tv-logo">
      </VToolbarTitle>

      <!-- Left menu items -->
      <VToolbarItems class="hidden-xs-only">
        <VBtn
          v-for="item in leftMenuItems"
          v-if="!item.shouldDisplay || item.shouldDisplay()"
          :key="item.title"
          :to="item.path"
          :nuxt="true"
          text
        >
          {{ item.title }}
        </VBtn>
      </VToolbarItems>
      <div class="d-none d-sm-flex d-md-none app-nav-right-items">
        <VMenu open-on-hover>
          <template v-slot:activator="{ on }">
            <VBtn
              v-on="on"
              text
            >
              More <font-awesome-icon :icon="['fal','chevron-down']" class="more-icon" />
            </VBtn>
          </template>

          <VList>
            <VListItem
              v-for="item in rightMenuItems"
              v-if="!item.shouldDisplay || item.shouldDisplay()"
              :key="item.title"
              @click=""
              :to="item.path"
              :nuxt="true"
              text
            >
              {{ item.title }}
            </VListItem>
          </VList>
        </VMenu>
      </div>
      <!-- Right menu items -->
      <VToolbarItems class="hidden-sm-and-down app-nav-right-items">
        <VBtn
          v-for="item in rightMenuItems"
          v-if="!item.shouldDisplay || item.shouldDisplay()"
          :key="item.title"
          :to="item.path"
          :nuxt="true"
          text
        >
          {{ item.title }}
        </VBtn>
      </VToolbarItems>
    </VAppBar>
  </div>
</template>

<script>
export default {
  props: {
    transparentAtTop: Boolean
  },
  data () {
    return {
      scrollDistance: 0, // Number of px scrolled down
      sidebar: false, // Whether the sidebar is open. Should be false by default
      socials: [
        { title: 'YouTube', path: 'https://youtube.com/rpitv', icon: 'youtube' },
        { title: 'Twitter', path: 'https://twitter.com/rpitv', icon: 'twitter' },
        { title: 'Reddit', path: 'https://reddit.com/u/rpi_tv', icon: 'reddit-alien' }
      ],
      leftMenuItems: [
        { title: 'Home', path: '/', icon: 'house' },
        { title: 'Productions', path: '/productions', icon: 'camera-movie' },
        { title: 'Contact Us', path: '/contact', icon: 'file-signature' },
        { title: 'About', path: '/about', icon: 'info-circle' }
      ],
      rightMenuItems: [
        { title: 'Join The Club', path: '/join', icon: 'hands-helping' },
        { title: this.$store.state.rcs_id + '@rpi.edu',
          path: '#',
          icon: 'user-circle',
          shouldDisplay: () => this.$store.getters.isAuthenticated },
        { title: 'Donate', path: '/donate', icon: 'donate', shouldDisplay: () => !this.$store.getters.isAuthenticated },
        { title: 'Login', path: '/login', icon: 'sign-in', shouldDisplay: () => !this.$store.getters.isAuthenticated },
        { title: 'Logout', path: '/logout', icon: 'sign-out', shouldDisplay: () => this.$store.getters.isAuthenticated }
      ]
    }
  },
  mounted () {
    window.addEventListener('scroll', this.scrollHandler)
    this.scrollHandler()
  },
  destroyed () {
    window.removeEventListener('scroll', this.scrollHandler)
  },
  methods: {
    scrollHandler () {
      this.scrollDistance = window.scrollY
    }
  }
}
</script>

<style lang="scss" scoped>

  /* Drawer */
  .drawer-item {
    padding-left: 15px;
    .drawer-icon {
      --fa-primary-color: #b05454;
      --fa-secondary-color: #f56e6c;
      --fa-secondary-opacity: 1.0;
    }
    .drawer-item-content {
      display: inline;
    }
  }
  .drawer-header .tv-logo {
    width: 75px;
    margin-right: 10px;
  }
  .drawer-social {
    position: relative;
    margin-bottom: 10px;
    top: -10px;
    left: 10px;

    * {
      margin-right: 10px;
    }
  }

  /* Main nav bar */
  .app-nav-right-items {
    position: absolute;
    right: 15px;
  }
  .more-icon {
    margin-left: 5px;
  }
  .app-nav-header .tv-logo {

    transition: opacity 0.5s;
    @media(min-width: 600px) and (max-width: 700px) {
      height: 30px !important;
    }
    @media (max-width: 1049px) {
      height: 60px;
      float: left;
      margin-right: 15px;
    }
    @media (min-width: 1050px), (max-width: 600px) {
      pointer-events: none;
      position: absolute;
      height: 150px;
      left: calc(50% - 100px);
      top: -20px;
      z-index: 3000;
    }
  }
</style>
