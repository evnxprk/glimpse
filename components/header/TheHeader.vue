<template>
  <div>
    <!-- Desktop nav bar -->
    <VAppBar
      :color="scrollDistance > 0 || !$props.transparentAtTop ? this.$vuetify.theme.currentTheme.primary.darken1 : '#00000000'"
      app
      elevate-on-scroll
      dark
    >
      <span class="hidden-sm-and-up" aria-label="open navigation drawer">
        <font-awesome-icon :icon="['fal', 'bars']" @click="sidebar = !sidebar" class="drawer-icon" />
      </span>

      <VToolbarTitle class="app-nav-header">
        <img src="../../assets/rpitv_logo_400.png" alt="Logo" class="tv-logo">
      </VToolbarTitle>

      <!-- Left menu items -->
      <VToolbarItems class="hidden-xs-only">
        <div v-for="item in menuItems.LEFT" :key="item.title" :class="'item-wrapper' + (item.customClass || '')">
          <HeaderButton
            v-if="(!item.sublist || item.sublist.length === 0) && (!item.conditional || item.conditional())"
            :class="item.customClass || ''"
            :title="item.title"
            :path="item.path"
          />
          <HeaderDropdown
            v-else-if="item.sublist && item.sublist.length > 0 && (!item.conditional || item.conditional())"
            :title="item.title"
            :class="item.customClass || ''"
            :items="item.sublist"
          />
        </div>
      </VToolbarItems>

      <div class="app-nav-right-items">
        <!-- Right menu items -->
        <VToolbarItems class="hidden-xs-only">
          <div v-for="item in menuItems.RIGHT" :key="item.title" :class="'item-wrapper ' + (item.customClass || '')">
            <HeaderButton
              v-if="(!item.sublist || item.sublist.length === 0) && (!item.conditional || item.conditional())"
              :class="item.customClass || ''"
              :title="item.title"
              :path="item.path"
            />
            <HeaderDropdown
              v-else-if="item.sublist && (!item.conditional || item.conditional())"
              :title="item.title"
              :class="item.customClass || ''"
              :items="item.sublist"
            />
          </div>
        </VToolbarItems>
      </div>
    </VAppBar>
  </div>
</template>

<script>
import HeaderButton from '~/components/header/HeaderButton'
import HeaderDropdown from '~/components/header/HeaderDropdown'
export default {
  components: { HeaderDropdown, HeaderButton },
  props: {
    transparentAtTop: Boolean
  },
  data () {
    return {
      scrollDistance: 0, // Number of px scrolled down
      sidebar: false, // Whether the sidebar is open. Should be false by default
      menuItems: {
        'LEFT': [
          {
            'title': 'Home',
            'path': '/',
            'icon': ['fal', 'house'],
            'sublist': [],
            'customClass': '',
            'conditional': null
          },
          {
            'title': 'Productions',
            'path': '/productions',
            'icon': ['fal', 'camera-movie'],
            'sublist': [],
            'customClass': '',
            'conditional': null
          },
          {
            'title': 'Contact Us',
            'path': '/contact',
            'icon': ['fal', 'file-signature'],
            'sublist': [],
            'customClass': '',
            'conditional': null
          },
          {
            'title': 'About',
            'path': '/about',
            'icon': ['fal', 'info-circle'],
            'sublist': [],
            'customClass': '',
            'conditional': null
          }
        ],
        'RIGHT': [
          {
            'title': 'Join The Club',
            'path': '/join',
            'icon': ['fal', 'hands-helping'],
            'sublist': [],
            'customClass': 'hidden-sm-and-down',
            'conditional': () => !this.$store.getters.isAuthenticated
          },
          {
            'title': 'Donate',
            'path': '/donate',
            'icon': ['fal', 'donate'],
            'sublist': [],
            'customClass': 'hidden-sm-and-down',
            'conditional': () => !this.$store.getters.isAuthenticated
          },
          {
            'title': 'Login',
            'path': '/login',
            'icon': ['fal', 'sign-in'],
            'sublist': [],
            'customClass': 'hidden-sm-and-down',
            'conditional': () => !this.$store.getters.isAuthenticated
          },
          {
            'title': this.$store.state.rcs_id + '@rpi.edu',
            'path': '#',
            'icon': ['fal', 'user-circle'],
            'sublist': [
              {
                'title': 'Account Settings',
                'path': '/account',
                'icon': ['fal', 'cog'],
                'sublist': [],
                'customClass': '',
                'conditional': null
              },
              {
                'title': 'Admin Panel',
                'path': '/admin',
                'icon': ['fal', 'toolbox'],
                'sublist': [],
                'customClass': '',
                'conditional': () => this.$store.state.admin
              }
            ],
            'customClass': 'hidden-sm-and-down',
            'conditional': () => this.$store.getters.isAuthenticated
          },
          {
            'title': 'Logout',
            'path': '/logout',
            'icon': ['fal', 'sign-out'],
            'sublist': [],
            'customClass': 'hidden-sm-and-down',
            'conditional': () => this.$store.getters.isAuthenticated
          },
          {
            'title': 'More',
            'path': '#',
            'icon': null,
            'sublist': [
              {
                'title': 'Join The Club',
                'path': '/join',
                'icon': ['fal', 'hands-helping'],
                'sublist': [],
                'customClass': '',
                'conditional': () => !this.$store.getters.isAuthenticated
              },
              {
                'title': 'Donate',
                'path': '/donate',
                'icon': ['fal', 'donate'],
                'sublist': [],
                'customClass': '',
                'conditional': () => !this.$store.getters.isAuthenticated
              },
              {
                'title': 'Login',
                'path': '/login',
                'icon': ['fal', 'sign-in'],
                'sublist': [],
                'customClass': '',
                'conditional': () => !this.$store.getters.isAuthenticated
              },
              {
                'title': this.$store.state.rcs_id + '@rpi.edu',
                'path': '#',
                'icon': ['fal', 'user-circle'],
                'sublist': [
                  {
                    'title': 'Account Settings',
                    'path': '/account',
                    'icon': ['fal', 'cog'],
                    'sublist': [],
                    'customClass': '',
                    'conditional': null
                  },
                  {
                    'title': 'Admin Panel',
                    'path': '/admin',
                    'icon': ['fal', 'toolbox'],
                    'sublist': [],
                    'customClass': '',
                    'conditional': () => this.$store.state.admin
                  }
                ],
                'conditional': () => this.$store.getters.isAuthenticated
              },
              {
                'title': 'Logout',
                'path': '/logout',
                'icon': ['fal', 'sign-out'],
                'sublist': [],
                'customClass': '',
                'conditional': () => this.$store.getters.isAuthenticated
              }
            ],
            'customClass': 'hidden-md-and-up',
            'conditional': null
          }
        ]
      }
    }
  },
  mounted () {
    window.addEventListener('scroll', this.scrollHandler)
    this.scrollHandler()
  },
  destroyed () {
    window.removeEventListener('scroll', this.scrollHandler)
  },
  methods: {
    scrollHandler () {
      this.scrollDistance = window.scrollY
    }
  }
}
</script>

<style lang="scss" scoped>

  .item-wrapper {
    display: flex;
    margin: 0;
  }
  /* Main nav bar */
  .app-nav-right-items {
    position: absolute;
    right: 15px;
    height: 100%;
  }

  .app-nav-header .tv-logo {

    transition: opacity 0.5s;
    @media(min-width: 600px) and (max-width: 700px) {
      height: 30px !important;
    }
    @media (max-width: 1049px) {
      height: 60px;
      float: left;
      margin-right: 15px;
    }
    @media (min-width: 1050px), (max-width: 600px) {
      pointer-events: none;
      position: absolute;
      height: 150px;
      left: calc(50% - 100px);
      top: -20px;
      z-index: 3000;
    }
  }
</style>
