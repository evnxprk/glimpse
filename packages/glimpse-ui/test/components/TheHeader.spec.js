import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'
import Sinon from 'sinon'
import VuetifyOptions from '../../vuetify.config'
import TheHeader from '~/components/header/TheHeader.vue'

Vue.use(VueRouter)
Vue.use(Vuex)
Vue.use(Vuetify)
const vuetifyInstance = new Vuetify(VuetifyOptions)

describe('TheHeader (Unauthenticated)', () => {
  let wrapper

  function mount (opts) {
    wrapper = shallowMount(TheHeader, {
      ...opts,
      vuetify: vuetifyInstance,
      router: new VueRouter(),
      stubs: {
        NuxtLink: RouterLinkStub
      } })
  }

  let store
  beforeEach(() => {
    store = new Vuex.Store({
      getters: {
        isAuthenticated: () => false
      }
    })
  })

  test('Properly renders', () => {
    mount({ store })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('headerbutton-stub[path=\'/login\']').attributes().title).toBe('Login')
  })

  test('Handles opaque top correctly', () => {
    mount({ store })
    expect(wrapper.find('vappbar-stub').element).toHaveAttribute('color', '#b05454')
  })

  test('Handles transparent top correctly', () => {
    mount({
      store,
      propsData: {
        transparentAtTop: true,
        attachToDocument: true
      }
    })
    expect(wrapper.find('vappbar-stub').element).toHaveAttribute('color', '#00000000')
  })

  test('Destroy unregisters event listener', () => {
    const spy = Sinon.stub()
    mount({
      store,
      attachToDocument: true,
      destroyed () {
        spy()
      }
    })
    wrapper.destroy()

    expect(spy.calledOnce).toBe(true)
  })
})

describe('TheHeader (Authenticated User)', () => {
  let wrapper

  function mount (opts) {
    wrapper = shallowMount(TheHeader, {
      ...opts,
      vuetify: vuetifyInstance,
      router: new VueRouter(),
      stubs: {
        NuxtLink: RouterLinkStub
      } })
  }

  let store
  beforeEach(() => {
    store = new Vuex.Store({
      getters: {
        isAuthenticated: () => true
      }
    })
  })

  test('Properly renders', () => {
    mount({ store })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('headerbutton-stub[path=\'/logout\']').attributes().title).toBe('Logout')
  })
})

describe('TheHeader (Authenticated Admin)', () => {
  let wrapper

  function mount (opts) {
    wrapper = shallowMount(TheHeader, {
      ...opts,
      vuetify: vuetifyInstance,
      router: new VueRouter(),
      stubs: {
        NuxtLink: RouterLinkStub
      } })
  }

  let store
  beforeEach(() => {
    store = new Vuex.Store({
      getters: {
        isAuthenticated: () => true
      },
      state: {
        admin: true
      }
    })
  })

  test('Properly renders', () => {
    mount({ store })
    expect(wrapper.element).toMatchSnapshot()
  })
})
