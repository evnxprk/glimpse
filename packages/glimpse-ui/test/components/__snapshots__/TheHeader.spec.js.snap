// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader (Authenticated Admin) Properly renders 1`] = `
<div>
  <mobileheader-stub
    class="hidden-sm-and-up"
    items="[object Object]"
  />
   
  <vappbar-stub
    app="true"
    color="#b05454"
    dark="true"
    elevateonscroll="true"
    extensionheight="48"
    src=""
    tag="header"
    tile="true"
    value="true"
  >
    <span
      aria-label="open navigation drawer"
      class="hidden-sm-and-up"
    >
      <font-awesome-icon-stub
        class="drawer-icon"
        icon="fal,bars"
      />
    </span>
     
    <vtoolbartitle-stub
      class="app-nav-header"
    >
      <img
        alt="Logo"
        class="tv-logo"
        src="../../assets/rpitv_logo_400.png"
      />
    </vtoolbartitle-stub>
     
    <vtoolbaritems-stub
      class="hidden-xs-only"
    >
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/"
          title="Home"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/productions"
          title="Productions"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/contact"
          title="Contact Us"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/about"
          title="About"
        />
      </div>
    </vtoolbaritems-stub>
     
    <div
      class="app-nav-right-items"
    >
      <vtoolbaritems-stub
        class="hidden-xs-only"
      >
        <div
          class="item-wrapper hidden-sm-only"
        >
          <!---->
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <!---->
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <!---->
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <headerdropdown-stub
            class="hidden-sm-only"
            depth="0"
            items="[object Object],[object Object]"
            title="undefined@rpi.edu"
          />
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <headerbutton-stub
            class="hidden-sm-only"
            path="/logout"
            title="Logout"
          />
        </div>
        <div
          class="item-wrapper hidden-md-and-up hidden-xs-only"
        >
          <headerdropdown-stub
            class="hidden-md-and-up hidden-xs-only"
            depth="0"
            items="[object Object],[object Object],[object Object],[object Object],[object Object]"
            title="More"
          />
        </div>
      </vtoolbaritems-stub>
    </div>
  </vappbar-stub>
</div>
`;

exports[`TheHeader (Authenticated User) Properly renders 1`] = `
<div>
  <mobileheader-stub
    class="hidden-sm-and-up"
    items="[object Object]"
  />
   
  <vappbar-stub
    app="true"
    color="#b05454"
    dark="true"
    elevateonscroll="true"
    extensionheight="48"
    src=""
    tag="header"
    tile="true"
    value="true"
  >
    <span
      aria-label="open navigation drawer"
      class="hidden-sm-and-up"
    >
      <font-awesome-icon-stub
        class="drawer-icon"
        icon="fal,bars"
      />
    </span>
     
    <vtoolbartitle-stub
      class="app-nav-header"
    >
      <img
        alt="Logo"
        class="tv-logo"
        src="../../assets/rpitv_logo_400.png"
      />
    </vtoolbartitle-stub>
     
    <vtoolbaritems-stub
      class="hidden-xs-only"
    >
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/"
          title="Home"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/productions"
          title="Productions"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/contact"
          title="Contact Us"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/about"
          title="About"
        />
      </div>
    </vtoolbaritems-stub>
     
    <div
      class="app-nav-right-items"
    >
      <vtoolbaritems-stub
        class="hidden-xs-only"
      >
        <div
          class="item-wrapper hidden-sm-only"
        >
          <!---->
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <!---->
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <!---->
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <headerdropdown-stub
            class="hidden-sm-only"
            depth="0"
            items="[object Object],[object Object]"
            title="undefined@rpi.edu"
          />
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <headerbutton-stub
            class="hidden-sm-only"
            path="/logout"
            title="Logout"
          />
        </div>
        <div
          class="item-wrapper hidden-md-and-up hidden-xs-only"
        >
          <headerdropdown-stub
            class="hidden-md-and-up hidden-xs-only"
            depth="0"
            items="[object Object],[object Object],[object Object],[object Object],[object Object]"
            title="More"
          />
        </div>
      </vtoolbaritems-stub>
    </div>
  </vappbar-stub>
</div>
`;

exports[`TheHeader (Custom) Properly renders 1`] = `
<div>
  <mobileheader-stub
    class="hidden-sm-and-up"
    items="[object Object]"
  />
   
  <vappbar-stub
    app="true"
    color="#b05454"
    dark="true"
    elevateonscroll="true"
    extensionheight="48"
    src=""
    tag="header"
    tile="true"
    value="true"
  >
    <span
      aria-label="open navigation drawer"
      class="hidden-sm-and-up"
    >
      <font-awesome-icon-stub
        class="drawer-icon"
        icon="fal,bars"
      />
    </span>
     
    <vtoolbartitle-stub
      class="app-nav-header"
    >
      <img
        alt="Logo"
        class="tv-logo"
        src="../../assets/rpitv_logo_400.png"
      />
    </vtoolbartitle-stub>
     
    <vtoolbaritems-stub
      class="hidden-xs-only"
    >
      <div
        class="item-wrapper sample-class-left1"
      >
        <headerbutton-stub
          class="sample-class-left1"
          path="/samplel1"
          title="Sample Left #1"
        />
      </div>
      <div
        class="item-wrapper sample-class-left2"
      >
        <headerbutton-stub
          class="sample-class-left2"
          path="/samplel2"
          title="Sample Left #2"
        />
      </div>
    </vtoolbaritems-stub>
     
    <div
      class="app-nav-right-items"
    >
      <vtoolbaritems-stub
        class="hidden-xs-only"
      >
        <div
          class="item-wrapper sample-r1"
        >
          <headerbutton-stub
            class="sample-r1"
            path="/sampler1"
            title="SampleRight #1"
          />
        </div>
        <div
          class="item-wrapper sample-r2"
        >
          <headerbutton-stub
            class="sample-r2"
            path="/sampler2"
            title="SampleRight #2"
          />
        </div>
        <div
          class="item-wrapper sample-r3"
        >
          <!---->
        </div>
      </vtoolbaritems-stub>
    </div>
  </vappbar-stub>
</div>
`;

exports[`TheHeader (Unauthenticated) Properly renders 1`] = `
<div>
  <mobileheader-stub
    class="hidden-sm-and-up"
    items="[object Object]"
  />
   
  <vappbar-stub
    app="true"
    color="#b05454"
    dark="true"
    elevateonscroll="true"
    extensionheight="48"
    src=""
    tag="header"
    tile="true"
    value="true"
  >
    <span
      aria-label="open navigation drawer"
      class="hidden-sm-and-up"
    >
      <font-awesome-icon-stub
        class="drawer-icon"
        icon="fal,bars"
      />
    </span>
     
    <vtoolbartitle-stub
      class="app-nav-header"
    >
      <img
        alt="Logo"
        class="tv-logo"
        src="../../assets/rpitv_logo_400.png"
      />
    </vtoolbartitle-stub>
     
    <vtoolbaritems-stub
      class="hidden-xs-only"
    >
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/"
          title="Home"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/productions"
          title="Productions"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/contact"
          title="Contact Us"
        />
      </div>
      <div
        class="item-wrapper "
      >
        <headerbutton-stub
          class=""
          path="/about"
          title="About"
        />
      </div>
    </vtoolbaritems-stub>
     
    <div
      class="app-nav-right-items"
    >
      <vtoolbaritems-stub
        class="hidden-xs-only"
      >
        <div
          class="item-wrapper hidden-sm-only"
        >
          <headerbutton-stub
            class="hidden-sm-only"
            path="/join"
            title="Join The Club"
          />
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <headerbutton-stub
            class="hidden-sm-only"
            path="/donate"
            title="Donate"
          />
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <headerbutton-stub
            class="hidden-sm-only"
            path="/login"
            title="Login"
          />
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <!---->
        </div>
        <div
          class="item-wrapper hidden-sm-only"
        >
          <!---->
        </div>
        <div
          class="item-wrapper hidden-md-and-up hidden-xs-only"
        >
          <headerdropdown-stub
            class="hidden-md-and-up hidden-xs-only"
            depth="0"
            items="[object Object],[object Object],[object Object],[object Object],[object Object]"
            title="More"
          />
        </div>
      </vtoolbaritems-stub>
    </div>
  </vappbar-stub>
</div>
`;
